### DISCUSSION TITLE
Read and process from stdin/a socket/... line by line

### DISCUSSION THREAD

**Discussion by robaca** [nushell/nushell#9144] [Q&A]
I have a use case where my script needs to parse lines from a socket continuously. In bash it would look like this:

```
while read -r line;
do
    # do something for each incoming line
    echo "$line" ;
done
```

How can I achive the same inside a nushell script?

**Comment by fdncred** [comment]
I'm not sure exactly but `open` works with streams, so maybe you can use that? e.g. `open /dev/random` will go on forever but not all commands work well with streams, some collect, which would be a problem here probably.

We have `input` but for reading stdin but I'm not sure it will work with what you're wanting to do.

**Reply by ghost** [reply]
input does not work with streams:
```
$ yes | input | lines | each {|i| print $i}
Error: nu::shell::io_error

  × I/O error
  help: Input/output error (os error 5)
```

you can just pipe it directly instead: `yes | lines | each {|i| print $i}`

in case `open` has some issues using `cat` instead might resolve the issue
